ownCloud/Client/Actions/Actions+Extensions/CopyAction.swift

```diff
@@ -69,11 +104,54 @@ class CopyAction : Action {
 		viewController.present(pickerNavigationController, animated: true)
 	}
 
-	override class func iconForLocation(_ location: OCExtensionLocationIdentifier) -> UIImage? {
-		if location == .moreItem || location == .moreDetailItem || location == .moreFolder || location == .contextMenuItem {
-			return UIImage(named: "copy-file")
+	func copyToPasteboard() {
+		guard context.items.count > 0, let viewController = context.viewController, let core = self.core, let tabBarController = viewController.tabBarController as? ClientRootViewController else {
+			completed(with: NSError(ocError: .insufficientParameters))
+			return
 		}
 
-		return nil
+		let items = context.items
+		items.forEach({ (item) in
+			// Internal Pasteboard
+			if let fileData = item.serializedData() {
+				let pasteboard = UIPasteboard(name: UIPasteboard.Name(rawValue: "com.owncloud.pasteboard"), create: true)
+				pasteboard?.setData(fileData as Data, forPasteboardType: "com.owncloud.uti.ocitem.copy")
+				tabBarController.pasteboardChangedCounter = UIPasteboard.general.changeCount
+			}
+
+			// General system-wide Pasteboard
+			if item.type == .collection {
+				let pasteboard = UIPasteboard.general
+				tabBarController.pasteboardChangedCounter = pasteboard.changeCount
```

Comment:
@hosy curious, why `UITabController` instance is used to keep track of paste board change count?

Reply:
I moved that code to the Key-Value store.